ndcube/ndcube.py

```diff
@@ -1034,6 +1035,43 @@ def __neg__(self):
         return self._new_instance(data=-self.data)
 
     def __add__(self, value):
+        kwargs = {}
+        if isinstance(value, NDData) and value.wcs is None:
+            if self.unit is not None and value.unit is not None:
+                    value_data = (value.data * value.unit).to_value(self.unit)
+            elif self.unit is None:
+                value_data = value.data
+            else:
+                raise TypeError("Cannot add unitless NDData to a unitful NDCube.")
+
+            # combine the uncertainty
+            if self.uncertainty is not None and value.uncertainty is not None:
+                new_uncertainty = self.uncertainty.propagate(
+                    np.add, value.uncertainty, result_data = value.data, correlation=0
+                )
+                kwargs["uncertainty"] = new_uncertainty
+            elif self.uncertainty is not None:
+                new_uncertainty = self.uncertainty
+                kwargs["uncertainty"] = new_uncertainty
+            elif value.uncertainty is not None:
+                new_uncertainty = value.uncertainty
+            else:
+                new_uncertainty = None
+
+            # combine mask
+            self_ma = np.ma.MaskedArray(self.data, mask=self.mask)
+            value_ma = np.ma.MaskedArray(value_data, mask=value.mask)
+
+            # addition
+            result_ma = self_ma + value_ma
+
+            # extract new mask and new data
+            kwargs["mask"] = result_ma.mask
+            kwargs["data"] = result_ma.data
+
+            # return the new NDCube instance
+            return self._new_instance(**kwargs)
```

Comment:
Move this line to the [end of the method](https://github.com/sunpy/ndcube/pull/794/files#diff-929794bb532be1486907b360285433b432f8505ab7e7ff63eca3eab657f7d00dR1091) and use the `kwargs` approach when handling the other cases, e.g. `Quantity`.  So, for example, [L1082](https://github.com/sunpy/ndcube/pull/794/files#diff-929794bb532be1486907b360285433b432f8505ab7e7ff63eca3eab657f7d00dR1082) would become:
```python
kwargs["data"] = self.data + value.to_value(cube_unit)
```